### PR TITLE
S96.10 Add Twitter Ads.

### PR DESCRIPTION
**Pull request by swordbeta**

### CODE REVIEW

**Comment by expand-dev** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [MediaPlatformAuthManagerFactory.php#L59](https://github.com/ExpandOnline/AuthManager/blob/28fd57a34f6f9c27baa5fb24540b5dafb0f535eb/Lib/MediaPlatformAuthManagerFactory.php#L59): Define and throw a dedicated exception instead of using a generic one. [rule](http://localhost:9000/sonarqube/coding_rules#rule_key=php%3AS112)

**Comment by expand-dev** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [MediaPlatformAuthManagerFactory.php#L59](https://github.com/ExpandOnline/AuthManager/blob/7743800de01bfdb56755683008e2100297c0fa74/Lib/MediaPlatformAuthManagerFactory.php#L59): Define and throw a dedicated exception instead of using a generic one. [rule](http://localhost:9000/sonarqube/coding_rules#rule_key=php%3AS112)

File: Lib/TwitterAds/TwitterAdsAuthManager.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	/**
+	 * Get the authentication url to add an user.
+	 *
+	 * @return string
+	 */
+	public function getAuthUrl() {
+		$tokens = $this->codebird->oauth_requestToken([
+			'oauth_callback' => $this->_getCallbackUrl(MediaPlatform::TWITTER_ADS)
+		]);
+		CakeSession::write('oauth_token_secret', $tokens->oauth_token_secret);
```
Comment (ronoudgenoeg) [inline Lib/TwitterAds/TwitterAdsAuthManager.php]: Uh? OAuth secrets in de sessie?

Comment (ronoudgenoeg) [inline Lib/TwitterAds/TwitterAdsAuthManager.php]: Dit proces lijkt nogal anders te gaan dan bij Google/FB?

Comment (swordbeta) [inline Lib/TwitterAds/TwitterAdsAuthManager.php]: @ronoudgenoeg Yep, Google/FB/Bing Ads/noem het maar op gebruikt OAUTH2, Twitter gebruikt OAUTH1a wat net wat anders is..

**Pull request merged**